server/gokbg3/grails-app/controllers/org/gokb/rest/JobsController.groovy

```diff
@@ -29,112 +30,92 @@ class JobsController {
-        int groupId = params.int('curatoryGroup')
+      // by linked Component
+      if (params.linkedItem) {
+        long compId = params.long('linkedItem')
 
         if (params.boolean('archived') == true) {
           result.data = []
-          def hqlTotal = JobResult.executeQuery("select count(jr.id) from JobResult as jr where jr.groupId = ?", [groupId.toLong()])[0]
-          def jobs = JobResult.executeQuery("from JobResult as jr where jr.groupId = ? order by jr.startTime desc", [groupId.toLong()], [max: max, offset: offset])
-
-          jobs.each { j ->
-            def component = j.linkedItemId ? KBComponent.get(j.linkedItemId) : null
-            // No JsonObject for list view
-
-            result.data << [
-              uuid: j.uuid,
-              description: j.description,
-              type: j.type ? [id: j.type.id, name: j.type.value, value: j.type.value] : null,
-              linkedItem: (component ? [id: component.id, type: component.niceName, uuid: component.uuid, name: component.name] : null),
-              startTime: j.startTime,
-              endTime: j.endTime,
-              status: j.statusText
-            ]
-          }
-
-          result['_pagination'] = [
-            offset: offset,
-            limit: max,
-            total: hqlTotal
-          ]
+          filterJobResults('linkedItemId', compId, max, offset, result)
         }
         else {
-          result = concurrencyManagerService.getGroupJobs(groupId, max, offset)
+          concurrencyManagerService.getComponentJobs(compId, max, offset).each { k, v ->
+            result.k = v
+          }
         }
       }
-      else {
-        result.result = 'ERROR'
-        response.setStatus(403)
-        result.message = "Insuffictient permissions to retrieve the jobs for this group"
-      }
+      // all jobs
     }
-    else if (user.superUserStatus) {
+    if (user.superUserStatus) {
       if (params.archived == "true") {
         result.data = []
         def hqlTotal = JobResult.executeQuery("select count(jr.id) from JobResult as jr")[0]
         def jobs = JobResult.executeQuery("from JobResult as jr order by jr.startTime desc", [], [max: max, offset: offset])
 
-        jobs.each { j ->
+        jobs.each { Job j ->
           def component = j.linkedItemId ? KBComponent.get(j.linkedItemId) : null
           // No JsonObject for list view
 
           result.data << [
-            uuid: j.uuid,
-            description: j.description,
-            type: j.type ? [id: j.type.id, name: j.type.value, value: j.type.value] : null,
-            linkedItem: (component ? [id: component.id, type: component.niceName, uuid: component.uuid, name: component.name] : null),
-            startTime: j.startTime,
-            endTime: j.endTime,
-            status: j.statusText
+              owner      : User.get(j.ownerId),
+              group      : CuratoryGroup.get(j.groupId),
```

Comment:
We don't want to serialize the complete objects here (including user password and all users linked to the group etc)! In any case, the owner should not be included at all. Similar to the `linkedItem`, a Map of selected properties has to be mapped manually for the `group`.

Reply:
did that